### PR TITLE
test clm versions

### PR DESCRIPTION
**Pull request by myaser**
[zalando-incubator/cluster-lifecycle-manager/#585](https://github.com/zalando-incubator/cluster-lifecycle-manager/pull/585)

### CODE REVIEW

**Comment by linki** [comment]
This fails here: https://github.com/zalando-incubator/cluster-lifecycle-manager/pull/585/files#diff-1c0e78391dde9bfd898c19603d220e4cbb3fcabd1a4727e8b990ff95cc68ccf7R242

It fails because it cannot fetch `Provisioners` because there's no such CRD. We cannot skip this part (e.g. by stopping a line before when `p.cluster.KarpenterPools() == 0`) because we need to go into this for removing the last karpenter pool.

It doesn't matter for testing karpenter so we can also find a hacky work-around for this.

File: test/e2e/Dockerfile [hunk 1]
```diff
@@ -24,7 +24,7 @@ RUN chmod +x /usr/bin/kubectl
 COPY --from=builder /go/bin/ginkgo /usr/local/bin/ginkgo
 
 # copy CLM
-COPY --from=registry.opensource.zalan.do/teapot/cluster-lifecycle-manager:latest /clm /usr/bin/clm
+COPY --from=registry.opensource.zalan.do/teapot/cluster-lifecycle-manager-test:pr-585-26 /clm /usr/bin/clm
```
Comment (myaser) [inline test/e2e/Dockerfile]:
```suggestion
COPY --from=registry.opensource.zalan.do/teapot/cluster-lifecycle-manager-test:pr-585-32 /clm /usr/bin/clm
```

**Comment by myaser** [comment]
> This fails here: https://github.com/zalando-incubator/cluster-lifecycle-manager/pull/585/files#diff-1c0e78391dde9bfd898c19603d220e4cbb3fcabd1a4727e8b990ff95cc68ccf7R242
> 
> It fails because it cannot fetch `Provisioners` because there's no such CRD. We cannot skip this part (e.g. by stopping a line before when `p.cluster.KarpenterPools() == 0`) because we need to go into this for removing the last karpenter pool.
> 
> It doesn't matter for testing karpenter so we can also find a hacky work-around for this.

fixed

File: test/e2e/Dockerfile [hunk 2]
```diff
@@ -24,7 +24,7 @@ RUN chmod +x /usr/bin/kubectl
 COPY --from=builder /go/bin/ginkgo /usr/local/bin/ginkgo
 
 # copy CLM
-COPY --from=registry.opensource.zalan.do/teapot/cluster-lifecycle-manager:latest /clm /usr/bin/clm
+COPY --from=registry.opensource.zalan.do/teapot/cluster-lifecycle-manager-test:pr-585-32 /clm /usr/bin/clm
```
Comment (myaser) [inline test/e2e/Dockerfile]:
```suggestion
COPY --from=registry.opensource.zalan.do/teapot/cluster-lifecycle-manager-test:pr-585-41 /clm /usr/bin/clm
```

File: test/e2e/Dockerfile [hunk 3]
```diff
@@ -24,7 +24,7 @@ RUN chmod +x /usr/bin/kubectl
 COPY --from=builder /go/bin/ginkgo /usr/local/bin/ginkgo
 
 # copy CLM
-COPY --from=registry.opensource.zalan.do/teapot/cluster-lifecycle-manager:latest /clm /usr/bin/clm
+COPY --from=registry.opensource.zalan.do/teapot/cluster-lifecycle-manager-test:pr-585-41 /clm /usr/bin/clm
```
Comment (myaser) [inline test/e2e/Dockerfile]:
```suggestion
COPY --from=registry.opensource.zalan.do/teapot/cluster-lifecycle-manager-test:pr-652-3 /clm /usr/bin/clm
```

**Pull request closed without merging**